src/compare_component_and_website_data/tmp.py

```diff
@@ -0,0 +1,23 @@
+"""Compare"""
+def clean(ls):
+    """Reads lines and cleans out common issues"""
+    return set([
+        x.replace('\\u00a0', '')
+        .replace('\xa0', '')
+        .replace('\n', '')
+        for x in ls])
+
+
+with open('comp2.txt', 'r') as f:
+    c_lines = f.readlines()  # 5,781
+    c_set = clean(c_lines)  # 2,331
+
+with open('json2.txt', 'r') as f:
+    j_lines = f.readlines()  # 5,781
+    j_set = clean(j_lines)  # 2,331
+
+source_diff = c_lines != j_lines  # True
+i = c_set.intersection(j_set)  # 2,332 100%
+c_diff = c_set.difference(j_set)  # 0
+j_diff = j_set.difference(c_set)  # 0
+print()
```

Comment:
## Analysis comparing component to website
### Results
Nico [asked me to compare](https://github.com/monarch-initiative/icd11/issues/10#issuecomment-2017919271) these two sources of information. I first created 2 sets of files:

(1) From the website https://www.orphadata.com/alignments/
I downloaded [en_product1.json.tar.gz](https://www.orphadata.com/data/json/en_product1.json.tar.gz) which contains mappings for ICD-10, ICD-11, OMIM, UMLS, MeSH, MedDRA and GARD.
I did `cat mappings-indent.json| grep '"Source": "ICD-11"' -B 1 > json.txt`
Then, I did some manual stripping of things until I had a plain text file with 1 ICD11 term on each line, saving as `json2.txt`.

(2) From `components/ordo.owl`
I did `cat components/ordo.owl | grep "<oboInOwl:hasDbXref>ICD-11:" > comp.txt`
Then, I did some manual stripping of things until I had a plain text file with 1 ICD11 term on each line, saving as `comp2.txt`.

Then, In this `tmp.py`, I read in both files, cleaned some unicode characters, and did a set difference on the terms.

I was very surprised to see two things:
1. The total number of lines in each of my source files `json2.txt` and `comp2.txt` was the same: 5,781
2. There was no set difference between the two. They contained exactly the same terms.

I double checked to see if there was something wrong with my Python code. I checked the files, and surely enough they did have the same number of lines. I checked visually and saw that the lines themselves were different, and confirmed that with `source_diff = c_lines != j_lines  # True`.

### What does this mean

Reply:
I looked a little more.

## How to know equivalent mappings
Example from `components/ordo.owl`:
```owl
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://www.orpha.net/ORDO/Orphanet_100081"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasDbXref"/>
        <owl:annotatedTarget>ICD-11:2B92.1</owl:annotatedTarget>
        <oboInOwl:source xml:lang="en">- E (Exact mapping: the two concepts are equivalent).
- Specific code (ICD-10/ICD-11: ORPHAcode has its own code in the targeted terminology).</oboInOwl:source>
    <owl:Axiom>
```